pkg/resource/deploy/builtins.go

```diff
@@ -286,7 +286,13 @@ func (p *builtinProvider) readStackReference(inputs resource.PropertyMap) (resou
 
 	outputs, err := p.backendClient.GetStackOutputs(p.context, name.StringValue())
 	if err != nil {
-		return nil, err
+		var decryptionErr *StackOutputDecryptionError
```

Comment:
Feel like it would be cleaner to thread the diag sink into GetStackOutputs and just write the diangostics there rather than having this StackOutputDecryptionError type. It's a bit odd in Go to return a result _and_ an error, it happens, but it feels like something that will get tripped up on.

Reply:
I started with that but it felt a bit uglier in terms of dependency graph, but I agree it's not like this approach is particularly beautiful either. Will give it another spin.

Reply:
I guess if we do that, there's no way (not that we need it now, and possibly not ever) to observe if secret decryption failed -- unless you can deeply introspect a `diag.Sink`, which I don't think you can (you could wrap one I guess)